src/AzureIoTHub.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Arduino. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <time.h>
+#include <sys/time.h>
+
+#if defined(ARDUINO_SAMD_FEATHER_M0)
+#include <Adafruit_WINC1500.h>
+#elif defined(ARDUINO_SAMD_ZERO) || defined(ARDUINO_SAMD_MKR1000)
+#include <WiFi101.h>
+#elif defined(ARDUINO_ARCH_ESP8266)
+#include <ESP8266WiFi.h>
+#else
+#error "Unsupported platform"
+#endif
+
+#include "util/NTPClient.h"
+
+#include "AzureIoTHub.h"
+
+AzureIoTHubClass AzureIoTHub;
+
+#if defined(ARDUINO_SAMD_FEATHER_M0)
```

Comment:
@ladyada there is potential to remove use of ```Adafruit_WINC1500``` library in this library and just use ```WiFi101``` if the variant for the "Adafruit Feather M0 WiFi" board defines following WINC1500 pin values like the MKR1000: https://github.com/arduino/ArduinoCore-samd/blob/master/variants/mkr1000/variant.h#L137-L143